pkg/tinydtls/contrib/sock_dtls.c

```diff
@@ -818,8 +818,8 @@ void sock_dtls_init(void)
 static void _ep_to_session(const sock_udp_ep_t *ep, session_t *session)
 {
     session->port = ep->port;
-    session->size = sizeof(ipv6_addr_t) +       /* addr */
-                    sizeof(unsigned short);     /* port */
```

Comment:
Strange: this seems to be there for no reason (extra code added and documented, that is just not correct),
it always was there since this pkg was added and tinydtls never changed the definition of that member.
(maybe there was an out of git version)

Reply:
So, I found it strange, and my guess is that we brought this bug since before we adapted the package to sock:

https://github.com/RIOT-OS/RIOT/blob/530e699de1eedff1b3d435b92398c2ed73a55f51/examples/dtls-echo/dtls-server.c#L117

https://github.com/RIOT-OS/RIOT/blob/530e699de1eedff1b3d435b92398c2ed73a55f51/examples/dtls-echo/dtls-client.c#L361

I'll fix those as well

Reply:
Done in cd980d4

Reply:
@miri64 solved that (for contiki the same is done)  - addr is not just the device address but the complete location of the other end